src/lib/parser/parser.ts

```diff
@@ -104,113 +76,108 @@ export class AuctionParser {
+				segment: preprocessedMessage.substring(
+					last_range.start,
+					range.start,
+				),
+				price: undefined,
+			});
+			last_range = range;
+		}
+		matches.push({
+			item: preprocessedMessage.substring(
+				last_range.start,
+				last_range.end,
+			),
+			segment: preprocessedMessage.substring(last_range.start),
+			price: undefined,
+		});
+
+		// Default to WTS or whichever is earlier in the message
 		let currentAuctionType = AuctionTypes.WTS;
-		let unknownItemString = '';
-
-		function addUnknownItemToResultsAndReset() {
-			// Check if unknownItemString contains more than 2 alphabetical characters excluding 'PP' or 'KPP'
-			if (
-				!/[a-zA-Z]{3,}/.test(
-					unknownItemString.replace(/\bPP\b|\bKPP\b/g, ''),
-				)
-			) {
-				// If not, reset unknownItemString and return early
-				unknownItemString = '';
-				return;
+		const wtb_match = preprocessedMessage.match(/WTB/);
+		const wts_match = preprocessedMessage.match(/WTS/);
+		if (wtb_match && wts_match) {
+			// eslint-disable-next-line @typescript-eslint/ban-ts-comment -- linter is wrong here
+			// @ts-ignore -- if both ANDs pass, then both are defined
+			if (wtb_match.index < wts_match.index) {
+				currentAuctionType = AuctionTypes.WTB;
 			}
+		} else if (wtb_match) {
+			currentAuctionType = AuctionTypes.WTB;
+		}
 
-			// if there is actualy content to parse, split the unknownItemString by delimiters
-			const sections = unknownItemString.split(/[,./|\-_+~<>]/);
-			sections.forEach((section) => {
-				// Trim any whitespace from the section
-				const trimmedSection = section.trim();
-				if (!trimmedSection) return;
-
-				// Attempt to parse the price from the last word in the section
-				const words = trimmedSection.split(/\s+/);
-				const lastWord = words[words.length - 1];
-				let price = undefined;
-				let item = trimmedSection;
-
-				// Check if last word is a number possibly followed by k or kpp
-				if (/^\d+(\.\d+)?k?(pp)?$/.test(lastWord)) {
-					price = parseFloat(lastWord.replace(/[^0-9.]/g, ''));
-					if (/k(pp)?$/.test(lastWord)) {
-						price *= 1000;
-					}
-					// Remove the price part from the item string
-					item = words.slice(0, -1).join(' ');
-				}
+		// Go through each segment and categorize it as WTB/WTS, and add price data
+		for (const segment of matches) {
+			if (segment.segment.match(/WTS/)) {
+				currentAuctionType = AuctionTypes.WTS;
+			} else if (segment.segment.match(/WTB/)) {
+				currentAuctionType = AuctionTypes.WTB;
+			}
 
-				// Add to buying or selling array
-				if (currentAuctionType === AuctionTypes.WTS) {
-					selling.push({ item, price });
-				} else {
-					buying.push({ item, price });
+			// Figure out price
+			const price_match = Array.from(
+				segment.segment.matchAll(
+					/(?<price>[0-9]+(\.[0-9]+)?)(?<kpp>k?p{0,2})/gi,
+				),
+			);
+			if (price_match.length > 0) {
+				const single_price_match = price_match[price_match.length - 1];
+				let price: number = +single_price_match[1];

```

Comment:
Annoyingly, the regex makes nice named groups, but TypeScript won't let me use them because they're "dynamic" and not defined clearly in the type for a RegexMatch. I'm not sure how to fix that...

Reply:
What are you wanting to do here? I may be able to help